frontend/optmath.lua

```diff
@@ -102,4 +104,25 @@ function Math.tmax(tab, func)
     return tmin_max(tab, func, "max")
 end
 
+--[[--
+Restricts a value within an interval.
+
+@tparam value value
+@tparma min minimum
+@tparam max maximum
+@treturn a value clipped to the interval [min, max]
+]]
+function Math.minmax(value, min, max)
+    if value <= min then
+        return min
+    elseif value >= max then
+        return max
+    end
+    return value
+end
+dbg:guard(Math, "minmax",
+    function(value, min, max)
+        assert(min < max, "Math.minmax: minimum must be less than maximum")
```

Comment:
And vice versa.

Reply:
??? Am I totally blind now?

Reply:
Maximum must be more than minimum, musn't it? :-)

Reply:
Isn't it the same?
Maximum must be more than minimum and
Minimum must be less than maximum. The latter is min<max and so more readable.

Reply:
I'm with @zwim - I don't see what else needs to be done :/
Please be more explicite: is this some logic missing ? Or you're just thinking about the string ?

Reply:
Talking about the string, yes. Maybe just:

```suggestion
        assert(min < max, "Math.minmax: minimum ("..min..") must be less than maximum ("..max..")")
```